docs/sphinx_api_docs_source/build_sphinx_api_docs.py

```diff
@@ -218,6 +219,8 @@ def _parse_and_process_html_to_mdx(
         title_str = title.get_text(strip=True)
         title_str = title_str.replace("#", "")
 
+        apply_markdown_adjustments(soup, html_file_path, html_file_contents)
```

Comment:
Can we return a copy of the values we're modifying instead of mutating the originals?

Reply:
I could do that change but I don't see a big difference there as the original files we're mutating are just temporary copies for generating the markdown, do you think it's worth it? Besides, I think it might feel a little bit inconsistent with the fact that the rest of the method is going to mutate them anyway and soup gets instantiated with the html content before the changes, so I could override the param but the instance would still be using the original one :thinking:

Reply:
I think you're right. normally you'd want to always copy a parameter and make the changes to that copy instead of the original but in this case it should be fine.